src/NHibernate/Dialect/Ingres9Dialect.cs

```diff
@@ -65,10 +70,30 @@ public override SqlString GetLimitString(SqlString queryString, SqlString offset
 			return pagingBuilder.ToSqlString();
 		}
 
+		public override string GetSequenceNextValString(string sequenceName)
+		{
+			return "select " + GetSelectSequenceNextValString(sequenceName) + " as seq";
+		}
+
+		public override string GetSelectSequenceNextValString(string sequenceName)
+		{
+			return "next value for " + sequenceName;
+		}
+
+		public override string GetCreateSequenceString(string sequenceName)
+		{
+			return "create sequence " + sequenceName;
+		}
+
+		public override string GetDropSequenceString(string sequenceName)
+		{
+			return "drop sequence " + sequenceName;
+		}
```

Comment:
Hibernate add after `sequenceName` the keyword `restrict`. Though [documented](http://docs.actian.com/ingres/11.0/index.html#page/SQLRef/08_SQLStatements.08.051.htm) only for other types of objects, it would be there for ignoring the drop if the sequence is referenced by another object in the database.

A bit conservative I think, and not done for other dialects, so I think above syntax is ok as currently is.